enzyme/test/Enzyme/ProbProg/condition.ll

```diff
@@ -0,0 +1,238 @@
+; RUN: %opt < %s %loadEnzyme -enzyme -O1 -S | FileCheck %s
+
+@.str = private unnamed_addr constant [11 x i8] c"predict, 0\00", align 1
+@.str.1 = private unnamed_addr constant [2 x i8] c"m\00", align 1
+@.str.2 = private unnamed_addr constant [2 x i8] c"b\00", align 1
+@enzyme_condition = dso_local local_unnamed_addr global i32 0, align 4
+@enzyme_interface = dso_local local_unnamed_addr global i32 0, align 4
+
+define dso_local double @normal(double %mean, double %var) #0 {
+entry:
+  ret double 4.000000e+00 ; chosen by fair dice roll.
+                          ; guaranteed to be random.
+}
+
+define dso_local double @normal_pdf(double noundef %mean, double noundef %var, double noundef %x) local_unnamed_addr #1 {
+entry:
+  %div = fdiv double 1.000000e+00, %var
+  %mul = fmul double %div, 0x40040D931FF62705
+  %div1 = fdiv double %mean, %var
+  %sub = fsub double %x, %div1
+  %square = fmul double %sub, %sub
+  %mul3 = fmul double %square, -5.000000e-01
+  %call4 = tail call double @exp(double noundef %mul3) #6
+  %mul5 = fmul double %mul, %call4
+  ret double %mul5
+}
+
+declare double @exp(double noundef) local_unnamed_addr #2
+
+define dso_local double @normal_logpdf(double noundef %mean, double noundef %var, double noundef %x) #1 {
+entry:
+  %call = tail call double @normal_pdf(double noundef %mean, double noundef %var, double noundef %x)
+  %call1 = tail call double @log(double noundef %call) #6
+  ret double %call1
+}
+
+declare double @log(double noundef) local_unnamed_addr #2
+
+define dso_local double @calculate_loss(double noundef %m, double noundef %b, double* nocapture noundef readonly %data, i32 noundef %n) local_unnamed_addr #3 {
+entry:
+  %cmp19 = icmp sgt i32 %n, 0
+  br i1 %cmp19, label %for.body.preheader, label %for.cond.cleanup
+
+for.body.preheader:                               ; preds = %entry
+  %wide.trip.count = zext i32 %n to i64
+  br label %for.body
+
+for.cond.cleanup:                                 ; preds = %for.body, %entry
+  %loss.0.lcssa = phi double [ 0.000000e+00, %entry ], [ %3, %for.body ]
+  ret double %loss.0.lcssa
+
+for.body:                                         ; preds = %for.body.preheader, %for.body
```

Comment:
Can you make a simpler version here with only one input? Also can you replace fmuladd -> mul and add